src/core/execute.c

```diff
@@ -2196,6 +2196,54 @@ static int apply_working_directory(const ExecContext *context,
         return 0;
 }
 
+static int setup_keyring(Unit *u, const ExecParameters *p, uid_t uid, gid_t gid) {
+        key_serial_t keyring;
+
+        assert(u);
+        assert(p);
+
+        /* Let's set up a new kernel session keyring for all system services. This has the benefit that each service
+         * runs with its own keyring shared among all processes of the service. If we don't do this the keyring will be
+         * automatically created on-demand and then linked to the user keyring, we we'd rather avoid to have a
+         * per-service keyring. */
```

Comment:
This comment is confusing. First you say you create a new keyring for all system services. But then you say you create one per service. And finally that you do not want a per-service keyring? This appears to be invoked per-service, so it appears that the last part is wrong and the first should s/all/each/ ?